### PR TITLE
Add CI to publish beta versions

### PR DESCRIPTION
**Pull request by bidoubiwa**
In this PR: 

- Adds a specific Beta testing CI to ensure the tests passes on the specific docker image of Meilisearch containing the feature
- Adds a versions checker for the Beta
- Adds a publish condition to publish with the beta tag

fixes: #590

### CODE REVIEW

File: .github/scripts/beta-docker-version.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+package_json_version=$(grep '"version":' package.json | cut -d ':' -f 2- | tr -d ' ' | tr -d '"' | tr -d ',')
+beta_feature=$(echo $package_json_version | sed -r 's/[0-9]+.[0-9]+.[0-9]+-//')
+beta_feature=$(echo $beta_feature | sed -r 's/-beta\.[0-9]*$//')
```
Comment (alallema) [inline .github/scripts/beta-docker-version.sh:5]: Why do it in 2 lines? It's a bit confusing, isn't it?

File: .github/scripts/beta-docker-version.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+docker_image=$(curl https://hub.docker.com/v2/repositories/getmeili/meilisearch/tags | jq | grep "$beta_feature" | head -1)
+docker_image=$(echo $docker_image | grep '"name":' | cut -d ':' -f 2- | tr -d ' ' | tr -d '"' | tr -d ',')
```
Comment (alallema) [inline .github/scripts/beta-docker-version.sh:8]: Same question

**Review by alallema — COMMENTED** [review]
Great addition !!!

File: .github/scripts/beta-docker-version.sh (hunk 1 above)
Comment (bidoubiwa) [reply .github/scripts/beta-docker-version.sh:5]: I think it helps in readability

File: .github/scripts/beta-docker-version.sh (hunk 2 above)
Comment (bidoubiwa) [reply .github/scripts/beta-docker-version.sh:8]: I think it helps in readability

**Review by alallema — APPROVED** [review]
LGTM! 🍗

**Comment by bidoubiwa** [comment]
bors merge

**Pull request merged**